src/lib/management/pages/StudentsPage/EditWeekDialog.js

```diff
@@ -0,0 +1,104 @@
+// NewWeekDialog.js
+import React from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, TextField, Button, FormControlLabel, Checkbox } from '@mui/material';
+import UrlInputField from '../../../../ui/UrlInputField/UrlInputField';
+
+const EditWeekDialog = ({ open, onClose, onAddWeek,onUpdateWeek, newWeek, setNewWeek }) => {
+  const handleInputChange = (event) => {
+    const { name, value } = event.target;
+    setNewWeek((prevWeek) => ({
+      ...prevWeek,
+      [name]: value,
+    }));
+  };
+
+  const handleToggleVisibility = (event) => {
+    setNewWeek((prevWeek) => ({
+      ...prevWeek,
+      is_visible: event.target.checked,
+    }));
+  };
+
+  return (
+    <Dialog open={open} onClose={onClose}>
+      <DialogTitle>{newWeek.id ? "Edit Week" : "Create a New Week"}</DialogTitle>
+      <DialogContent>
+        <TextField
+          autoFocus
+          margin="dense"
+          name="title"
+          label="Week Title"
+          fullWidth
+          value={newWeek.title}
+          onChange={handleInputChange}
+          size="small"
+        />
+        <TextField
+          margin="dense"
+          name="subject"
+          label="Subject"
+          fullWidth
+          value={newWeek.subject}
+          onChange={handleInputChange}
+          size="small"
+        />
+        <TextField
+          margin="dense"
+          name="order_num"
+          label="Order Number"
+          fullWidth
+          value={newWeek.order_num || ''}
```

Comment:
you can know how many weeks there are from the father component
and then give a default value of weeks.length